### PR TITLE
Add plot types component to the town status screen

### PR DESCRIPTION
**Pull request by Altruiis**
#### Description: 

This PR adds a plot type component in the same vein as the resident component to the town status screen. The plots and their counts are shown by hovering (using the `RESIDENTOWNED` cache map), while clicking displays the same output as `/town plots`. Internally, it allows access to the TownBlockType cache Maps, to allow for retrieving plot counts easily. Also, I added two strings to the `en_US.yml` file but I wasn't sure whether to update the version or not, so let me know what I should do there.
____
#### New Nodes/Commands/ConfigOptions: 

```
status_plot_string: 'Plots'
status_plot_hover: '%s:'
```
____
#### Relevant Towny Issue ticket:

#5890

____
- [x] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

____
#### Images
image
image

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 1]
```diff
@@ -4,19 +4,7 @@
...
-import com.palmergames.bukkit.towny.object.ResidentList;
-import com.palmergames.bukkit.towny.object.SpawnLocation;
-import com.palmergames.bukkit.towny.object.Town;
-import com.palmergames.bukkit.towny.object.TownBlock;
-import com.palmergames.bukkit.towny.object.TownBlockType;
-import com.palmergames.bukkit.towny.object.TownyObject;
-import com.palmergames.bukkit.towny.object.TownyWorld;
-import com.palmergames.bukkit.towny.object.Translation;
-import com.palmergames.bukkit.towny.object.Translator;
+import com.palmergames.bukkit.towny.object.*;
```
Comment (Warriorrrr) [inline src/com/palmergames/bukkit/towny/TownyFormatter.java:7]: Undo import on demand

File: src/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 2]
```diff
@@ -40,11 +28,7 @@
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 import java.text.SimpleDateFormat;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Calendar;
-import java.util.List;
-import java.util.Locale;
+import java.util.*;
```
Comment (Warriorrrr) [inline src/com/palmergames/bukkit/towny/TownyFormatter.java:31]: Here too

File: src/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 3]
```diff
@@ -412,6 +396,18 @@ public static StatusScreen getStatus(Town town, CommandSender sender) {
...
+			screen.addComponentOf("plotsnewline", Component.newline());
+			TextComponent text = Component.empty();
+			Map<TownBlockType, Integer> cache = town.getTownBlockTypeCache().getCache(TownBlockTypeCache.CacheType.RESIDENTOWNED);
+			for (TownBlockType type : TownBlockTypeHandler.getTypes().values()) {
+				int value = cache.getOrDefault(type, 0);
+				text = text.append(Component.text(colourKeyValue(translator.of("status_plot_hover", type.getFormattedName()), String.valueOf(value))).append(Component.newline()));
+			}
+			text = text.append(Component.text(translator.of("status_hover_click_for_more")));
+			screen.addComponentOf("plots", colourHoverKey(translator.of("status_plot_string")), 
+				HoverEvent.showText(text), ClickEvent.runCommand("/towny:town plots"));
```
Comment (Warriorrrr) [inline src/com/palmergames/bukkit/towny/TownyFormatter.java:410]:
```suggestion
			screen.addComponentOf("plots", colourHoverKey(translator.of("status_plot_string")).hoverEvent(HoverEvent.showText(text)).clickEvent(ClickEvent.runCommand("/towny:town plots")));
```

**Review by Warriorrrr — CHANGES_REQUESTED** [review]

File: src/com/palmergames/bukkit/towny/TownyFormatter.java (hunk 3 above)
Comment (Altruiis) [reply src/com/palmergames/bukkit/towny/TownyFormatter.java:429]: I can't change that, colourHoverKey is a String. Other hover/click events are done the same way.

File: resources/lang/en-US.yml [hunk 4]
```diff
@@ -1677,4 +1677,9 @@ msg_err_could_not_find_any_towns_to_rtp: 'Could not find any open, public towns
...
 msg_cost_spawn_rtp: '&bYou were charged %s to teleport to a random town plot that is for sale.'
 #Message shown to a player when they use the town rtp command.
-msg_you_are_about_to_be_teleported_to_a_plot: 'You are about to be teleported to a plot which is forsale in %s which will let you join. Use ''/plot claim'' to claim your plot when you arrive.'
\ No newline at end of file
+msg_you_are_about_to_be_teleported_to_a_plot: 'You are about to be teleported to a plot which is forsale in %s which will let you join. Use ''/plot claim'' to claim your plot when you arrive.'
+
+#
+#Plots on Status Screen
+status_plot_string: 'Plots'
+status_plot_hover: '%s:'
```
Comment (LlmDl) [inline resources/lang/en-US.yml:1685]: Each language string should have its own comment line, which is imported into Crowdin separately.

File: src/com/palmergames/bukkit/towny/TownyFormatter.java [hunk 5]
```diff
@@ -412,6 +415,18 @@ public static StatusScreen getStatus(Town town, CommandSender sender) {
 					.append(Component.newline())
 					.append(Component.text(translator.of("status_hover_click_for_more")))),
 				ClickEvent.runCommand("/towny:town reslist "+ town.getName()));
+			
+			// Plots
+			screen.addComponentOf("plotsnewline", Component.newline());
+			TextComponent text = Component.empty();
+			Map<TownBlockType, Integer> cache = town.getTownBlockTypeCache().getCache(TownBlockTypeCache.CacheType.RESIDENTOWNED);
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownyFormatter.java:422]: This is only going to show player-owned plots and not the entire town's plot type collection. CacheType.ALL ought to be used.

**Review by LlmDl — CHANGES_REQUESTED** [review]
Couple of minor things. Nice first PR, you beat me to adding this myself.

File: src/com/palmergames/bukkit/towny/TownyFormatter.java (hunk 3 above)
Comment (Warriorrrr) [reply src/com/palmergames/bukkit/towny/TownyFormatter.java:429]: Ah right, legacy text

**Review by Warriorrrr — APPROVED** [review]

**Review by LlmDl — APPROVED** [review]

**Pull request merged**